src/domains/ecommerce-app/api/ecommerce-checkout/v3/_transaction_policy.xml.tpl

```diff
@@ -0,0 +1,106 @@
+<policies>
+    <inbound>
+        <base />
+        <!-- pass rptId value into header START -->
+        <set-header name="x-rpt-id" exists-action="delete" />
+        <set-variable name="paymentNotices" value="@(((JArray)((JObject)context.Request.Body.As<JObject>(preserveContent: true))["paymentNotices"]))" />
+        <set-variable name="rptIds" value="@{
+            string result = "";
+            foreach (JObject notice in ((JArray)(context.Variables["paymentNotices"]))) {
+                if( notice.ContainsKey("rptId") == true )
+                {
+                    result += notice["rptId"].Value<string>()+", ";
+                }
+            }
+            return result;
+        }" />
+        <choose>
+            <when condition="@((string)context.Variables["rptIds"] != "")">
+                <set-header name="x-rpt-id" exists-action="override">
+                    <value>@((string)context.Variables.GetValueOrDefault("rptIds",""))</value>
+                </set-header>
+            </when>
+        </choose>
+        <!-- pass rptId value into header END -->
+
+        <!-- Post Token PDV START-->
+        <send-request ignore-error="true" timeout="10" response-variable-name="pdv-token" mode="new">
+            <set-url>${pdv_api_base_path}/tokens</set-url>
+            <set-method>PUT</set-method>
+            <set-header name="x-api-key" exists-action="override">
+                <value>{{ecommerce-personal-data-vault-api-key}}</value>
+            </set-header>
+            <set-body>@{
+            JObject requestBody = (JObject)context.Request.Body.As<JObject>(true);
+            return new JObject(
+                    new JProperty("pii",  (string)requestBody["email"])
+                ).ToString();
+                }</set-body>
+        </send-request>
+        <choose>
+            <when condition="@(((IResponse)context.Variables["pdv-token"]).StatusCode != 200)">
+                <return-response>
+                    <set-status code="502" />
+                    <set-header name="Content-Type" exists-action="override">
+                        <value>application/json</value>
+                    </set-header>
+                    <set-body>
+                    {
+                        "title" : "Bad gateway - Invalid PDV response",
+                        "status":  502,
+                        "detail":  "Cannot tokenize fiscal code"
+                    }
+                    </set-body>
+                </return-response>
+            </when>
+        </choose>
+
+        <set-variable name="pdvToken" value="@(((IResponse)context.Variables["pdv-token"]).Body.As<JObject>())" />
+        <set-variable name="emailToken" value="@((string)((JObject)context.Variables["pdvToken"])["token"])" />
```

Comment:
at this level, we add tokenization of fiscalCode and pass it to backend (as header x-user-id)?

Reply:
Yes, should I do this in this pr or in a dedicated one?

Reply:
Ok, it's a dedicated PR :-)